### PR TITLE
[Task-3] Components in React

### PR DESCRIPTION
**Pull request by rogerosha**
**Acceptance Criteria for 1**
- [x] Mentor is added to pr
- [x] AC copied to pr description
- [x] Link to deployed version is added to pr
- [x] The ProductsList component correctly receives product data as props.
- [x] ProductCard component created and renders a single product.
- [x] Header and Footer elements are present.
- [x] ProductList displays right on mobile devices

**Acceptance Criteria for 1***
- [x] The Header component has clickable elements that toggle the display between the About and ProductsList components.
- [x] State management is handled in App.tsx, with state determining which component to show.
- [x] LocalStorage added to save items in card after page reload

Deployed page: https://rogerosha.github.io/fe-react-2024/

Thanks! ✨

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
+export type Page = 'about' | 'products';
+
 function App() {
+    const [products, setProducts] = useState([]);
+    const [page, setPage] = useState<Page>('products');
+
+    useEffect(() => {
+        const fetchProducts = async () => {
+            try {
+                const response = await fetch('https://ma-backend-api.mocintra.com/api/v1/products?limit=8&offset=2');
```
Comment (KPIcoder) [inline src/App.tsx:23]: I like the fact that you went ahead of the course and used the data from the api for the products 👍
You will learn a better approach for data fetching on the lesson about Redux

File: src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
+
+import { ProductsListComponent } from '@/components/productsList/ProductsList.component.tsx';
 
 import { AboutComponent } from './components/about/About.component.tsx';
 import { FooterComponent } from './components/footer/Footer.component.tsx';
 import { HeaderComponent } from './components/header/Header.component.tsx';
 
 import './App.css';
 
+export type Page = 'about' | 'products';
```
Comment (KPIcoder) [inline src/App.tsx]: there is no clean way to declare your pages unless you learn one of the routing libraries. But for now please move this type to a separate file in the directory where you declare your types. Keep your `App.tsx` file as clean as possible

File: src/App.tsx [hunk 3]
```diff
@@ -1,16 +1,41 @@
...
+
 function App() {
+    const [products, setProducts] = useState([]);
```
Comment (KPIcoder) [inline src/App.tsx]: do not forget to use the power of typescript
```suggestion
    const [products, setProducts] = useState<Product[]>([]);
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,16 +1,41 @@
...
+        };
+
+        fetchProducts();
+    }, []);
+
     return (
         <div>
-            <HeaderComponent />
-            <AboutComponent />
+            <HeaderComponent page={page} onPageClick={(newPage) => setPage(newPage)} />
```
Comment (KPIcoder) [inline src/App.tsx]: Try to minimise the logic in the JSX even the simple one. Move `onPageClick` to a separate function with a good and clear name. It is a good practice as it makes your code easier to read

**Review by KPIcoder — CHANGES_REQUESTED** [review]
- Fix the linter issues
- Extend the header on the About page to take all the width of the screen and stay consistent with the Products page
- Move your components that you use as pages to a separate folder `pages`
- Use local storage to save data if you want to receive 1*

File: src/components/productsList/ProductsList.component.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import styles from './productsList.module.css';
+
+export interface ProductsListComponentProps {
+    products: Product[];
+}
+
+export const ProductsListComponent: React.FC<ProductsListComponentProps> = ({ products }) => (
+    <div className={styles['products-list']}>
+        {products.map((product, index) => (
```
Comment (KPIcoder) [inline src/components/productsList/ProductsList.component.tsx]: As long as you don't have to remove / add new products you will not face any issues with using index as a key, but I will recommend to use `product.id` instead. Read more here https://react.dev/learn/rendering-lists#why-does-react-need-keys
```suggestion
        {products.map((product) => (
          <ProductCard key={product.id} product={product} />
```

**Comment by rogerosha** [comment]
> * Use local storage to save data if you want to receive 1*

I tried to do my best, but can't handle this task 😢

**Review by KPIcoder — APPROVED** [review]

**Pull request merged**